### ISSUE TITLE
Feature Request: Add support for parameter groups in optimizers

### ISSUE THREAD

**Issue by yuchaoran2011** [ml-explore/mlx#2837]
Currently, MLX optimizers only accept scalar hyperparameters, requiring all parameters to use the same learning rate, weight decay, etc. This differs from PyTorch's optimizer API, which supports parameter groups with different configurations.
```
optimizer = optim.AdamW(learning_rate=1e-4, weight_decay=0.01)
```
This means all parameters must use identical optimizer settings, which is problematic for common training practices like Applying weight decay only to weight matrices but not biases. 

I had to create multiple optimizers and manually split the gradients into decay groups and no-decay groups every training step. This approach introduces code complexity and is less efficient:
```
optimizer_decay = optim.AdamW(learning_rate=1e-4, weight_decay=0.01)
optimizer_skip_decay = optim.AdamW(learning_rate=1e-3, weight_decay=0.0)
def step(x, y):
    loss, grads = ...
    # Split gradients into two groups
    grads_to_decay, grads_to_skip_decay = {}, {}
    split_grads(grads, grads_to_decay, grads_to_skip_decay)  # Complex recursive logic
    
    # Clip and update separately
    clipped_grads_to_decay, _ = clip_grad_norm(grads_to_decay, max_norm=1.0)
    clipped_grads_to_skip_decay, _ = clip_grad_norm(grads_to_skip_decay, max_norm=1.0)
    optimizer_decay.update(model, clipped_grads_to_decay)
    optimizer_skip_decay.update(model, clipped_grads_to_skip_decay)
```
If we have support for parameter group like in PyTorch, the above code can be simplified to:
```
# Single optimizer with parameter groups
optimizer = optim.AdamW([
    {'params': decay_params, 'learning_rate': 1e-4, 'weight_decay': 0.01},
    {'params': no_decay_params, 'learning_rate': 1e-3, 'weight_decay': 0.0}
])

def step(x, y):
    loss, grads = ...
    clipped_grads, _ = clip_grad_norm(grads, max_norm=1.0)
    optimizer.update(model, clipped_grads) 
```
Please consider this improvement proposal. If there's any simpler way of doing it in MLX that I missed, feel free to suggest.

**Comment by CC-Yeh** [comment]
Maybe `MultiOptimizer` can help in your case?

https://ml-explore.github.io/mlx/build/html/python/optimizers/_autosummary/mlx.optimizers.MultiOptimizer.html#mlx.optimizers.MultiOptimizer

**Comment by yuchaoran2011** [comment]
@CC-Yeh How come I didn't find it earlier. That's exactly what I need. Thanks!
